dex/wait/queue.go

```diff
@@ -204,31 +203,27 @@ func (q *TaperingTickerQueue) Run(ctx context.Context) {
 			return
 		}
 
-		var nextTick time.Time
-		switch {
-		case w.tick <= fullSpeedTicks:
-			nextTick = time.Now().Add(q.fastestInterval)
-		case w.tick < fullyTapered: // ramp up the interval
-			prog := float64(w.tick-fullSpeedTicks) / (fullyTapered - fullSpeedTicks)
-			interval := q.fastestInterval + time.Duration(math.Round(prog*taper))
-			nextTick = time.Now().Add(interval)
-		default:
-			nextTick = time.Now().Add(q.slowestInterval)
-		}
-		if nextTick.After(w.Expiration) {
-			nextTick = w.Expiration
-		}
-
-		w.nextTick = nextTick
 		w.tick++
+		w.nextTick = nextTick(w.tick, q.slowestInterval, q.fastestInterval,
+			time.Now(), w.Expiration)
```

Comment:
pulled `nextTick` in a separate func to be able to unit-test it;

**note:** `w.tick++` now happens before next tick value is calculated; this is to make it fit with the 1-off change related to `fullSpeedTicks` I've mentioned above